app/org/sagebionetworks/bridge/services/SchedulePlanService.java

```diff
@@ -69,6 +83,23 @@ public SchedulePlan updateSchedulePlan(Study study, SchedulePlan plan) {
         // Plan must always be in user's study
         plan.setStudyKey(study.getIdentifier());
         
+        // Verify that all GUIDs that are supplied already exist in the plan. If they don't (or GUID is null),
+        // then add a new GUID.
+        SchedulePlan existing = schedulePlanDao.getSchedulePlan(study, plan.getGuid());
+        Set<String> existingGuids = getAllActivityGuids(existing);
+        
+        List<Schedule> schedules = plan.getStrategy().getAllPossibleSchedules();
+        for (Schedule schedule : schedules) {
+            for (int i=0; i < schedule.getActivities().size(); i++) {
+                Activity activity = schedule.getActivities().get(i);
+                if (activity.getGuid() == null || !existingGuids.contains(activity.getGuid())) {
```

Comment:
What is the reason for creating an activity with a new guid if that activity/guid is not in the plan? should an activity only belong to one schedule?